### ISSUE TITLE
Measure nonlinear OpenES sigma and learning-rate sensitivity

### ISSUE THREAD

**Issue by Ahmadfauzi34** [Ahmadfauzi34/ai_agent_builder#237]
## Purpose

Follow #233/#234. Search-budget evidence shows the 74-parameter nonlinear policy benefits much more from additional generations than from simply doubling population. Before changing OpenES algorithm semantics, measure whether fixed sigma or learning rate is driving the long champion plateaus / `NO_IMPROVEMENT` phases.

## Fixed workload

Keep unchanged:

- fresh installed Python wheel and typed host surface;
- policy `Linear(6->8)->ReLU->Linear(8->2)`;
- canonical parameter dimension 74;
- same deterministic 2D dynamics, disturbances, four scenarios, reward/cost terms, action clamp, and 16-step horizon;
- OpenES strategy 0 and seed `20_260_917`;
- population 8, generations 48 = exactly 384 candidate evaluations per case;
- Python owns environment/reward/schedule, graph owns policy execution, ES owns candidate generation/update;
- `ask_f32()` candidate windows feed `apply_flat` directly.

## One-variable-at-a-time matrix

Use independent optimizers and reset the graph to the same zero trainable state before every case:

```text
BASE:    sigma 0.15, learning_rate 0.06
S_LOW:   sigma 0.08, learning_rate 0.06
S_HIGH:  sigma 0.25, learning_rate 0.06
LR_LOW:  sigma 0.15, learning_rate 0.03
LR_HIGH: sigma 0.15, learning_rate 0.10
```

Do not tune both parameters simultaneously in this slice.

## Required evidence

For each case report:

- zero, initial champion, final champion rewards;
- baseline/training gain ratios;
- full champion and generation-best histories;
- optimizer flags;
- exact candidate evaluation count;
- number of champion improvements;
- longest consecutive champion plateau in generations;
- first/last improvement generation;
- count of `NO_IMPROVEMENT` flags;
- ask/apply/rollout/tell timing and total training wall time;
- finite fitness and stable batch/cardinality/identity proofs.

Use the validated case with the highest final champion reward only for exact ProgramBundle state/reward replay. Selection is evidence plumbing, not a production promotion rule.

## Interpretation

- sigma variants dominate BASE -> exploration radius is the immediate search-dynamics lever;
- learning-rate variants dominate -> update step size is the immediate lever;
- all variants remain close while plateaus persist -> fixed OpenES dynamics may be the next algorithm-level bottleneck, but gather evidence before changing semantics;
- stronger hyperparameters improve nonlinear policy but still do not match the simple Linear control policy -> task simplicity remains an important explanation and model-surface growth is still unjustified.

## Non-goals

No optimizer algorithm change, no adaptive sigma, no schedule/annealing, no ABI/facade change, no activation expansion, no graph/controller change, no Linear wrapper work, no Math Program v10, no NumPy/DLPack/PyO3, and no Go/C++ work.

Implement as a research-only slice after #234 is merged.
